retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
That seems very volatile and highly dependant on implementation details. (Next stop Regression City.)

A user not familiar with implementation details or without a concept of labels (and there are and will be many) should not be prevented from creating a basic setup. A descriptive mode will need a section in the docs, so will `honor_labels` as it is only self-explaining for about ~4 people (and even we are having a hard time to grasp the semantics).
Difference being that with `mode: "federate"` a user can infer what it's there for without understanding what it does under the hood (and he doesn't care and he shouldn't have to – just works[tm]).

Plus the flexibility stuff. I'm all for keeping the number of modes as low as possible.
Those are my thoughts - I'll leave the decision to anyone willing to break the tie.